core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+                        <span class="oppia-navbar-role-text" style="right: 4px;">A</span>
+                      {% elif is_moderator %}
+                        <span class="oppia-navbar-role-text">M</span>
+                      {% endif %}
+                    </div>
+                  {% endif %}
+                </div>
+              </a>
+
+              <ul class="dropdown-menu ng-cloak oppia-navbar-dropdown" role="menu"
+                  ng-mouseover="onMouseoverProfilePictureOrDropdown($event)"
+                  ng-mouseleave="onMouseoutProfilePictureOrDropdown($event)"
+                  ng-show="profileDropdownIsActive">
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/profile/{{username}}"
+                     class="protractor-test-profile-link">
+                    <strong>{{username}}</strong>
+                  </a>
+                </li>
+                <hr class="oppia-top-right-menu-item-separator">
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/dashboard"
+                     class="protractor-test-dashboard-link">
+                    <span translate="I18N_TOPNAV_DASHBOARD"></span>
+                  </a>
+                </li>
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/notifications_dashboard"
+                     class="protractor-test-notifications-link">
+                    <span translate="I18N_TOPNAV_NOTIFICATIONS"></span>
+                    <span ng-if="numUnseenNotifications > 0">
+                      (<[numUnseenNotifications]>)
+                    </span>
+                  </a>
+                </li>
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/preferences"
+                     class="protractor-test-preferences-link">
+                    <span translate="I18N_TOPNAV_PREFERENCES"></span>
+                  </a>
+                </li>
+
+                {% if is_moderator %}
+                  <li>
+                    <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/moderator" target="_blank">
+                      <span translate="I18N_TOPNAV_MODERATOR_PAGE"></span>
+                    </a>
+                  </li>
+                {% endif %}
+                {% if is_super_admin %}
+                  <li>
+                    <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/admin" target="_blank">
+                      <span translate="I18N_TOPNAV_ADMIN_PAGE"></span>
+                    </a>
+                  </li>
+                {% endif %}
+                <hr class="oppia-top-right-menu-item-separator">
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="{{logout_url}}">
```

Comment:
Does the entire logout URL need to be passed from the backend? Seems unnecessary to me.